SignalServiceKit/src/Loki/Protocol/Session Management/SessionManagementProtocol.swift

```diff
@@ -234,6 +242,8 @@ public final class SessionManagementProtocol : NSObject {
         let publicKey = thread.contactIdentifier()
         print("[Loki] End session message received from: \(publicKey).")
         // Notify the user
+        let masterHexEncodedPublicKey = storage.getMasterHexEncodedPublicKey(for: publicKey, in: transaction) ?? publicKey
+        let thread = TSContactThread.getOrCreateThread(withContactId: masterHexEncodedPublicKey, transaction: transaction)
```

Comment:
Why is this needed?

Reply:
To make sure the session reset in progress message is shown in the master thread. Or if this is triggered by a secondary device, the info message seems to be missing.

Reply:
That makes sense :)